content/influxdb/clustered/admin/licensing.md

```diff
@@ -0,0 +1,170 @@
+related:
+  - /influxdb/clustered/install/licensing/
+  - /influxdb/clustered/admin/upgrade/
+---
+
+Install and manage your InfluxDB Clustered license to authorize the use of
+the InfluxDB Clustered software.
+
+- [Install your InfluxDB license](#install-your-influxdb-license)
+- [Recover from a license misconfiguration](#recover-from-a-license-misconfiguration)
+- [Renew your license](#renew-your-license)
+- [License enforcement](#license-enforcement)
+  - [A valid license is required](#a-valid-license-is-required)
+  - [Periodic license checks](#periodic-license-checks)
+  - [License grace periods](#license-grace-periods)
+    - [License expiry logs](#license-expiry-logs)
+    - [Query brownout](#query-brownout)
+
+{{% note %}}
+#### License enforcement is currently an opt-in feature
+
+In currently available versions of InfluxDB Clustered, license enforcement is an
+opt-in feature. This allows InfluxData to introduce license enforcement to a
+limited number of customers in a way that can easily be disabled if issues arise.
+In the future, all releases of InfluxDB Clustered will require an active license
+to use the product.
+
+To opt into license enforcement, enable the `useLicensedBinaries` feature flag
+in your `AppInstance` resource _([See the example below](#enable-feature-flag))_.
+
+{{% /note %}}
+
+## Install your InfluxDB license
+
+{{% note %}}
+If setting up an InfluxDB Clustered deployment for the first time, first
+[set up the prerequisites](/influxdb/clustered/install/licensing/) and
+[configure your cluster](/influxdb/clustered/install/configure-cluster/).
+You will be able to install your license after the `influxdb` namespace is
+created and prepared.
+{{% /note %}}
+
+1.  If you haven't already,
+    [request an InfluxDB Clustered license](https://influxdata.com/contact-sales).
+2.  InfluxData provides you with a `license.yml` file that encapsulates your
+    license token as a custom Kubernetes resource.
+3.  Use `kubectl` to apply and create the `License` resource in your InfluxDB
+    namespace:
+
+    ```sh
+    kubectl apply --filename license.yml --namespace influxdb
+    ```
+
+4.  <span id="enable-feature-flag"></span>
+    Update your `AppInstance` resource to enable the `useLicensedBinaries` feature flag.
+    Add the `useLicensedBinaries` entry to the `.spec.package.spec.featureFlags`
+    property--for example:
+
+    ```yml
+    apiVersion: kubecfg.dev/v1alpha1
+    kind: AppInstance
+    # ...
+    spec:
+      package:
+        spec:
+          featureFlags:
+            - useLicensedBinaries
+    ```
+
+InfluxDB Clustered detects the `License` resource and extracts the credentials
+into a secret required by InfluxDB Clustered Kubernetes pods.
+Pods validate the license secret both at startup and periodically (roughly once
+per hour) while running.
+
+## Recover from a license misconfiguration
+
+If you deploy a licensed release of InfluxDB Clustered without a valid license,
```

Comment:
This paragraph isn't quite right. It's correct in describing the behavior of the new pods (ie they will crashloop) but it also implies that the existing pods will be fine. However, if the cluster doesn't have a valid, properly-installed license then even the existing pods will enter a `CrashLoopBackoff` state.

Reply:
> Is the license not included in the state(?) used to deploy the old pods? 

Licenses are accessed by a volume-mounted kubernetes `Secret`. All pods in the cluster share the same Secret. This k8s `Secret` is created/updated by the `license-controller` pod whenever the customer installs the `License` resource.

>If an invalid License resource is deployed, it will affect both old and new pods?

That's right! Worth noting that if the cluster initially tries deploying an invalid license the abovementioned `license-controller` pod will fail to verify it and decline to create the necessary `Secret`. The only way a license that is installed in the cluster should ever become invalid is for it to expire after initially being installed in a valid state.

> How does the customer recover from this? Just trigger a new deploy with a valid License resource?

Yes, deploying a new valid license is the answer to a bad one (eg it expired while installed in the cluster) causing a crash loop backoff.

>Is downtime inevitable if they deploy an invalid license?

I mentioned this in an earlier response, but it should be impossible for an invalid license to override an existing valid license. This is because the `license-controller` will detect the invalidity and refuse to overwrite the `Secret` mounted into the `IOx` pods.

> If so, is there a way to validate a license before deploying it?

Yeah. I know I'm repeating myself, but I want to make sure there isn't room for ambiguity. When the customer runs `kubectl apply -f license.yml`, the `License` CustomResource may be overwritten (we can't prevent that short of a solution like a validating k8s webhook, which I would prefer to avoid but isn't totally out of the question); however, the `license-controller` that watches for the `License` CustomResource will attempt to validate the license signature before creating/updating the `Secret` mounted into IOx pods. This validation prevents a bad license from overwriting a good one and causing the IOx pods to go down.

Reply:
Thank you! I make sure this is clear in this section.